spec/vindi/client_spec.rb

```diff
@@ -62,8 +62,12 @@
     describe 'errors' do
       it 'retuns not found error' do
         VCR.use_cassette("raise_404_error") do
-          expect{ client.get 'hello' }
-            .to raise_error Vindi::Error::NotFound
+          begin
+            client.get 'hello'
+            expect(client.last_response.status).to eq 404
+          rescue Vindi::Error::NotFound => error
+            expect(error.status_code).to eq 404
```

Comment:
Se a exceção não for lançada, este `expect` nunca será verificado. Recomendo manter a expectation anterior e criar um novo teste. Provavelmente exista outro jeito de fazer isso melhor com o Rspec:

```ruby
describe 'errors' do
  it 'raises an exception' do
    VCR.use_cassette("raise_404_error") do
      expect{ client.get 'hello' }.to raise_error Vindi::Error::NotFound
    end
  end

  it 'returns an error with status_code' do
    VCR.use_cassette("raise_404_error") do
      begin
        client.get 'hello'
      rescue Vindi::Error::NotFound => error
        expect(error.status_code).to eq(404)
      end
    end
  end
end
```

Reply:
Mas melhor mesmo, mesmo que se do jeito que eu fiz antes funcionasse.